ethcore/src/error.rs

```diff
@@ -220,6 +220,9 @@ pub enum Error {
 	/// A convenient variant for String.
 	#[display(fmt = "{}", _0)]
 	Msg(String),
+	/// State errors
+	#[display(fmt = "State error ({})", _0)]
+	State(account_state::Error),
```

Comment:
do we still need a `Error::Trie` if we have this one here?

Reply:
Hmm, if I remove it most compilation errors are indeed from state-related code but there are still a bunch from snapshot code that also pokes at the trie directly.

Reply:
I looked into this and it seems fairly involved. Changing `account-state` to use a standard `Result` with its own `Error` is easy but there's plenty of code in `ethcore` that returns `vm::Result`, which means that I need a `From<account_state::Error> for vm::Error` which obviously doesn't work.
I think the "fix" is to change `ethcore` to stop returning `vm::Result`. I'd rather not do that refactoring in this PR.

Reply:
(and besides, there's still `snapshot` that still needs use `TrieError` so we wouldn't get rid of it here anyway)